jerry-core/ecma/builtin-objects/ecma-builtin-atomics.c

```diff
@@ -0,0 +1,264 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "ecma-builtins.h"
+#include "ecma-globals.h"
+#include "ecma-helpers.h"
+#include "jrt.h"
+#include "ecma-atomics-object.h"
+
+#if JERRY_BUILTIN_ATOMICS
+
+#define ECMA_BUILTINS_INTERNAL
+
+#include "ecma-builtins-internal.h"
+
+/**
+ * This object has a custom dispatch function.
+ */
+#define BUILTIN_CUSTOM_DISPATCH
+
+/**
+ * List of built-in routine identifiers.
+ */
+enum
+{
+  ECMA_ATOMICS_ROUTINE_START = 0,
+  ECMA_ATOMICS_ROUTINE_ADD, /**< Atomics add routine */
+  ECMA_ATOMICS_ROUTINE_AND, /**< Atomics and routine */
+  ECMA_ATOMICS_ROUTINE_COMPAREEXCHANGE, /**< Atomics compare exchange routine */
+  ECMA_ATOMICS_ROUTINE_EXCHANGE, /**< Atomics exchange routine */
+  ECMA_ATOMICS_ROUTINE_ISLOCKFREE, /**< Atomics is lock free routine */
+  ECMA_ATOMICS_ROUTINE_LOAD, /**< Atomics load routine */
+  ECMA_ATOMICS_ROUTINE_OR, /**< Atomics or routine */
+  ECMA_ATOMICS_ROUTINE_STORE, /**< Atomics store routine */
+  ECMA_ATOMICS_ROUTINE_SUB, /**< Atomics sub routine */
+  ECMA_ATOMICS_ROUTINE_WAIT, /**< Atomics wait routine */
+  ECMA_ATOMICS_ROUTINE_NOTIFY, /**< Atomics notify routine */
+  ECMA_ATOMICS_ROUTINE_XOR, /**< Atomics xor routine */
+};
+
+#define BUILTIN_INC_HEADER_NAME "ecma-builtin-atomics.inc.h"
+#define BUILTIN_UNDERSCORED_ID atomics
+
+#include "ecma-builtin-internal-routines-template.inc.h"
+
+/** \addtogroup ecma ECMA
+ * @{
+ *
+ * \addtogroup ecmabuiltins
+ * @{
+ *
+ * \addtogroup atomics ECMA Atomics object built-in
+ * @{
+ */
+
+/**
+ * The Atomics object's 'compareExchange' routine
+ *
+ * See also:
+ *          24.4.4
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+static ecma_value_t
+ecma_builtin_atomics_compare_exchange (ecma_value_t arg1, /**< typedArray argument */
+                                       ecma_value_t arg2, /**< index argument */
+                                       ecma_value_t arg3,  /**< expectedValue argument */
+                                       ecma_value_t arg4) /**< replacementValue */
```

Comment:
There is no reason not to have normal argument names (argX does not mean anything)